src/wasm-s-parser.h

```diff
@@ -77,6 +77,11 @@ class Element {
   Element* setString(cashew::IString str__, bool dollared__, bool quoted__);
   Element* setMetadata(size_t line_, size_t col_, SourceLocation* startLoc_);
 
+  // comparisons
+  bool operator==(Name name) { return isStr() && str() == name; }
+
+  template<typename T> bool operator!=(T t) { return !(*this == t); }
```

Comment:
Is there an `operator==` besides the one you added here? If not, it would be simpler to not make this generic.

Reply:
Where do we need to use these operators? I thought `Element` as a more higher container-like entity than `Name`, so it feels a bit weird to me. If we don't use it in too many places then calling `str()` to compare names wouldn't be too bad..

Reply:
I think avoiding templates would have the additional benefit of improving error messages, but I don't feel too strongly about this either.

Reply:
It's not many places. But it is pretty nice to do `if (x == REF)` instead of `if (x.isStr() && x.str() == REF)` I think?

Reply:
Hmm, yeah, that doesn't look bad... I'm OK either way 😅